src/qibolab/platforms/abstract.py

```diff
@@ -134,24 +137,47 @@ def reload_settings(self):
         self.relaxation_time = settings["settings"]["repetition_duration"]
         self.sampling_rate = settings["settings"]["sampling_rate"]
 
-        # TODO: Create better data structures for native gates
+        # Load native gates
         self.native_gates = settings["native_gates"]
-        self.native_single_qubit_gates = self.native_gates["single_qubit"]
-        if "two_qubit" in self.native_gates:
-            self.native_two_qubit_gates = self.native_gates["two_qubit"]
-            for gates in self.native_gates["two_qubit"].values():
-                self.two_qubit_natives |= set(gates.keys())
-        else:
-            # dummy value to avoid transpiler failure for single qubit devices
-            self.two_qubit_natives = {"CZ"}
-
         # Load characterization settings and create ``Qubit`` and ``Channel`` objects
         for q in settings["qubits"]:
             if q in self.qubits:
                 for name, value in settings["characterization"]["single_qubit"][q].items():
                     setattr(self.qubits[q], name, value)
             else:
-                self.qubits[q] = Qubit(q, **settings["characterization"]["single_qubit"][q])
+                self.qubits[q] = qubit = Qubit(q, **settings["characterization"]["single_qubit"][q])
+                # register channels to qubits when we are using the old format
+                # needed for ``NativeGates`` to work
+                if "qubit_channel_map" in self.settings:
+                    ro, qd, qf, _ = self.settings["qubit_channel_map"][q]
+                    if ro is not None:
+                        qubit.readout = Channel(ro)
+                    if qd is not None:
+                        qubit.drive = Channel(qd)
+                    if qf is not None:
+                        qubit.flux = Channel(qf)
+                # register single qubit native gates to Qubit objects
+                qubit.native_gates = NativeSingleQubitGates.from_dict(qubit, self.native_gates["single_qubit"][q])
+        if "two_qubit" in self.native_gates:
+            for pair, gatedict in self.native_gates["two_qubit"].items():
+                qubit1, qubit2 = pair.split("-")
+                if qubit1.isdigit():
+                    qubit1 = int(qubit1)
+                if qubit2.isdigit():
+                    qubit2 = int(qubit2)
+                sequences = {n: NativeSequence.from_dict(n, self.qubits, seq) for n, seq in gatedict.items()}
+                self.two_qubit_natives[frozenset((qubit1, qubit2))] = sequences
+
+    @property
+    def two_qubit_native_types(self):
+        """Set with names of all different two qubit native gates.
+
+        Used in the gate-to-gate transpiler.
+        """
+        native_types = {name for gates in self.two_qubit_natives.values() for name in gates.keys()}
+        if not native_types:
```

Comment:
```suggestion
        if len(native_types) == 0:
```

If possible, avoid relying on falsiness of non-boolean, since it can be misleading (even though here it is for sure a set, so there is no doubt).